Fix - 23913, Delete product query transients when removing terms via …

aslamshekh opened the pull request:
#23913  I have delete product query transients when removing terms via wp_remove_object_terms() 
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

Closes #23913 .

Please review from your side and let me know if there is anything.

Commit:
Fix - 23913, Delete product query transients when removing terms via wp_remove_object_terms()

vedanshujain commented on includes/class-wc-post-data.php:
```diff
@@ -30,6 +30,7 @@ public static function init() {
 		add_action( 'shutdown', array( __CLASS__, 'do_deferred_product_sync' ), 10 );
 		add_action( 'set_object_terms', array( __CLASS__, 'force_default_term' ), 10, 5 );
 		add_action( 'set_object_terms', array( __CLASS__, 'delete_product_query_transients' ) );
+		add_action( 'delete_term_relationships', array( __CLASS__, 'delete_product_query_transients' ) );
```
this should be hooked to `deleted_term_relationships` instead. can you please change this and update. thanks

aslamshekh committed:
Update class-wc-post-data.php

Thanks for the feedback @vedanshujain. As per your suggestion, I have changed hooked to deleted_term_relationships. Please let me know if any.

vedanshujain reviewed (approved):
LGTM! Thanks.
